### PR TITLE
Fix/sub generators

### PR DESCRIPTION
**Pull request by sendilkumarn**
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green

This enables sub-generators to use configurations from the blueprint. 

Fixes: #10738

### CODE REVIEW

File: generators/ci-cd/index.js [hunk 1]
```diff
@@ -81,17 +81,18 @@ module.exports = class extends BaseGenerator {
             },
             getConfig() {
                 this.jhipsterVersion = packagejs.version;
-                this.baseName = this.config.get('baseName');
-                this.applicationType = this.config.get('applicationType');
-                this.skipClient = this.config.get('skipClient');
-                this.clientPackageManager = this.config.get('clientPackageManager');
-                this.buildTool = this.config.get('buildTool');
-                this.herokuAppName = this.config.get('herokuAppName');
+                const configuration = this.getAllJhipsterConfig(this, true);
```
Comment (MathieuAA) [inline generators/ci-cd/index.js]: I've seen this code somewhere, but I don't know why this work compared to what we already have!
What's the difference between these two approaches? thanks @sendilkumarn :)

**Comment by mshima** [comment]
I think this is not needed with https://github.com/jhipster/generator-jhipster/pull/10664

File: generators/ci-cd/index.js (hunk 1 above)
Comment (mshima) [reply generators/ci-cd/index.js]: getAllJhipsterConfig is a workaround for https://github.com/jhipster/generator-jhipster/issues/10663.
It gets all configs from every namespace.

**Comment by sendilkumarn** [comment]
@mshima I don't think so. For example in Kotlin all the server level keys will only be present in the sub-generator's configurations and not in main generator's configuration.

File: generators/ci-cd/index.js (hunk 1 above)
Comment (MathieuAA) [reply generators/ci-cd/index.js]: I think I get it, thanks :)

**Comment by mshima** [comment]
I don't know how Kotlin works.
So https://github.com/jhipster/generator-jhipster/pull/10664 will break Kotlin?

**Pull request merged**